### PR TITLE
Prep for 1.0 release

### PR DESCRIPTION
**Pull request by odow**
- [ ] This still requires a MOI 1.0 tag, but we can start the update now.

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -1,21 +1,16 @@
...
-CEnum = "fa961155-64e5-5f13-b03f-caf6b980ea82"
 GLPK_jll = "e8aa6df9-e6ca-548a-97ff-1f85fc5b8b98"
-Libdl = "8f399da3-3557-5675-b5ff-fb832c97cbdb"
 MathOptInterface = "b8f27783-ece8-5eb3-8dc8-9495eed66fee"
 
 [compat]
-BinaryProvider = "~0.5"
-CEnum = "0.3, 0.4"
 GLPK_jll = "=5.0.0, =5.0.1"
 MathOptInterface = "~0.10.5"
```
Comment (odow) [inline Project.toml:12]:
```suggestion
MathOptInterface = "1"
```

**Comment by odow** [comment]
Merging as-is because I'd like to work on #206 and #204 before we release v1.0.

**Comment by joaquimg** [comment]
Why not allowing external binaries?
People might need to stick with some slightly different version or their own patch of GLPK.
I do this a lot for external binaries (never did for GLPK though).
This question is probably broader this might not be the best venue (was even merged while I was writing XD ).

**Comment by odow** [comment]
As part of JuMP 1.0, I'll publicize this a lot more: https://jump.dev/JuMP.jl/stable/developers/custom_solver_binaries/

There's now a standard way of using custom binaries, so we don't need these hacks in each solver.

**Comment by joaquimg** [comment]
awesome!

**Pull request merged**